diff -ruN forge/mcp/src/minecraft/net/minecraft/entity/monster/EntityCreeper.java baseEdits_777/mcp/src/minecraft/net/minecraft/entity/monster/EntityCreeper.java
--- forge/mcp/src/minecraft/net/minecraft/entity/monster/EntityCreeper.java	2013-07-09 03:42:53.130368000 +0300
+++ baseEdits_777/mcp/src/minecraft/net/minecraft/entity/monster/EntityCreeper.java	2013-07-09 04:30:41.259006500 +0300
@@ -19,6 +19,7 @@
 import net.minecraft.item.Item;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.DamageSource;
+import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
 public class EntityCreeper extends EntityMob
@@ -259,4 +260,25 @@
         super.onStruckByLightning(par1EntityLightningBolt);
         this.dataWatcher.updateObject(17, Byte.valueOf((byte)1));
     }
+    
+        
+    /**
+     * Called frequently so the entity can update its state every tick as required. For example, zombies and skeletons
+     * use this to react to sunlight and start to burn.
+     */
+    public void onLivingUpdate()
+    {
+        if (this.worldObj.isDaytime() && !this.worldObj.isRemote)
+        {
+            float f = this.getBrightness(1.0F);
+
+            if (f > 0.5F && this.rand.nextFloat() * 30.0F < (f - 0.4F) * 2.0F && this.worldObj.canBlockSeeTheSky(MathHelper.floor_double(this.posX), MathHelper.floor_double(this.posY), MathHelper.floor_double(this.posZ)))
+            {
+                this.setFire(8);
+            }
+        }
+
+        super.onLivingUpdate();
+    }
+
 }
